pkg/util/winutil/process.go

```diff
@@ -0,0 +1,276 @@
+// +build windows
+
+package winutil
+
+import (
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	modntdll                      = windows.NewLazyDLL("ntdll.dll")
+	modkernel                     = windows.NewLazyDLL("kernel32.dll")
+	procNtQueryInformationProcess = modntdll.NewProc("NtQueryInformationProcess")
+	procReadProcessMemory         = modkernel.NewProc("ReadProcessMemory")
+	procIsWow64Process            = modkernel.NewProc("IsWow64Process")
+)
+
+// C definition from winternl.h
+
+//typedef enum _PROCESSINFOCLASS {
+//    ProcessBasicInformation = 0,
+//    ProcessDebugPort = 7,
+//    ProcessWow64Information = 26,
+//    ProcessImageFileName = 27,
+//    ProcessBreakOnTermination = 29
+//} PROCESSINFOCLASS;
+
+type PROCESSINFOCLASS uint32
+
+const (
+	ProcessBasicInformation   = PROCESSINFOCLASS(0)
+	ProcessDebugPort          = PROCESSINFOCLASS(7)
+	ProcessWow64Information   = PROCESSINFOCLASS(26)
+	ProcessImageFileName      = PROCESSINFOCLASS(27)
+	ProcessBreakOnTermination = PROCESSINFOCLASS(29)
+)
+
+// IsWow64Process determines if the specified process is running under WOW64
+// that is, if it's a 32 bit process running on 64 bit winodws
+func IsWow64Process(h windows.Handle) (is32bit bool, err error) {
+	is32bit = false
+	err = nil
+	var wow64Process uint32
+
+	r, _, _ := procIsWow64Process.Call(uintptr(h),
+		uintptr(unsafe.Pointer(&wow64Process)))
+
+	if r == 0 {
+		return false, windows.GetLastError()
+	}
+	if wow64Process == 0 {
+		is32bit = false
+	} else {
+		is32bit = true
+	}
+	return
+}
+
+// NtQueryInformationProcess wraps the Windows NT kernel call of the same name
+func NtQueryInformationProcess(h windows.Handle, class PROCESSINFOCLASS, target, size uintptr) (err error) {
+	r, _, _ := procNtQueryInformationProcess.Call(uintptr(h),
```

Comment:
nit: in golang if all arguments are really short, they could just be in one line. There's technically no hard limit on how long a line should be.

Reply:
My preference is to leave it as is.  Especially when using the `syscall` interface, and there's multiple casts per argument, makes it much easier to read IMO